src/Html2Text.php

```diff
@@ -270,15 +271,34 @@ static function iterateOverNode($node, $in_pre = false) {
 		if (isset($node->childNodes)) {
 
 			$n = $node->childNodes->item(0);
+			$previousSiblingName = null;
 
 			while($n != null) {
 
-				$text = static::iterateOverNode($n, $in_pre || $name == 'pre');
+				$text = static::iterateOverNode($n, $previousSiblingName, $in_pre || $name == 'pre', $is_office_document);
 
-				$output .= $text;
+				// Pass current node name to next child, as previousSibling does not appear to get populated
+				switch(true) {
+					case $n instanceof \DOMDocumentType:
+					case $n instanceof \DOMProcessingInstruction:
+						$previousSiblingName = null;
+						break;
+					case $n instanceof \DOMText:
```

Comment:
This code was actually slightly wrong anyway--the first conditions shouldn't have nullified previousSiblingName as they are invisible and ignored. In any case, fixed now.